lib/facter/resolvers/augeas_resolver.rb

```diff
@@ -14,10 +14,28 @@ def post_resolve(fact_name)
         end
 
         def read_augeas_version(fact_name)
-          output = Facter::Core::Execution.execute('augparse --version 2>&1', logger: log)
-          @fact_list[:augeas_version] = Regexp.last_match(1) if output =~ /^augparse (\d+\.\d+\.\d+)/
+          @fact_list[:augeas_version] = read_augeas_from_cli
+          @fact_list[:augeas_version] ||= read_augeas_from_gem
+
           @fact_list[fact_name]
         end
+
+        def read_augeas_from_cli
+          output = Facter::Core::Execution.execute('augparse --version 2>&1', logger: log)
+          Regexp.last_match(1) if output =~ /^augparse (\d+\.\d+\.\d+)/
+        end
+
+        def read_augeas_from_gem
+          require 'augeas'
+
+          return ::Augeas.create { |aug| aug.get('/augeas/version') } if ::Augeas.singleton_method(:create)
```

Comment:
This will raise error if augeas 0.5.0 is installed.
```ruby
require 'augeas'
>> Gem.loaded_specs['ruby-augeas'].version   #=> #<Gem::Version "0.5.0">
>> ::Augeas.singleton_method(:create)
NameError (undefined singleton method `create' for `Augeas')
```

Reply:
You can branch on `Gem.loaded_specs` as there are two gems that can be installed and provide `Augeas`

ruby-augeas(has .open methdod) - https://rubygems.org/gems/ruby-augeas 
augeas(has create methods) - https://rubygems.org/gems/augeas

```ruby
if Gem.loaded_specs['augeas']
  create
else
  open
```

Reply:
Changed to `Gem.loaded_specs`